File: src/main/java/org/example/util/EncryptionUtil.java
package org.example.util;

import javax.crypto.KeyGenerator;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

public class EncryptionUtil {

    private EncryptionUtil() {
    }

    public static SecretKey generateKey() {
        try {
            KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
            keyGenerator.init(128);
            return keyGenerator.generateKey();
        } catch (NoSuchAlgorithmException e) {
            System.out.println("Algorithm not found");
            return null;
        }
    }


    public static String encodeKey(SecretKey key) {
        return Base64.getEncoder().encodeToString(key.getEncoded());
    }

    public static boolean isValidAESKey(String key) {
        try {
            byte[] decodedKey = Base64.getDecoder().decode(key);
            int keyLength = decodedKey.length;
            return keyLength == 16 || keyLength == 24 || keyLength == 32;
        } catch (IllegalArgumentException e) {
            return false;
        }
    }

    public static SecretKey decodeKey(String encodedKey) {
        byte[] decodedKey = Base64.getDecoder().decode(encodedKey);
        return new SecretKeySpec(decodedKey, 0, decodedKey.length, "AES");
    }

    public static void validateKey(String key) {
        if (key != null && !isValidAESKey(key)) {
            throw new IllegalArgumentException("Invalid AES key.");
        }

    }


}

File: src/main/java/org/example/util/EncryptionUtil.kt
package org.example.util

import org.bouncycastle.jce.provider.BouncyCastleProvider
import java.security.NoSuchAlgorithmException
import java.security.NoSuchProviderException
import java.security.Security
import java.util.*
import javax.crypto.KeyGenerator
import javax.crypto.SecretKey
import javax.crypto.spec.SecretKeySpec

object EncryptionUtil {
    init {
        Security.addProvider(BouncyCastleProvider())
    }

    fun generateKey(): SecretKey? {
        try {
            val keyGenerator = KeyGenerator.getInstance("AES", "BC")
            keyGenerator.init(128)
            return keyGenerator.generateKey()
        } catch (e: NoSuchAlgorithmException) {
            println("Algorithm not found❗")
            return null
        } catch (e: NoSuchProviderException) {
            println("Algorithm not found❗")
            return null
        }
    }

    fun encodeKey(key: SecretKey): String? {
        return Base64.getEncoder().encodeToString(key.getEncoded())
    }

    fun isValidAESKey(key: String?): Boolean {
        if (key == null) {
            return false
        }
        try {
            val decodedKey = Base64.getDecoder().decode(key)
            val keyLength = decodedKey.size
            return keyLength == 16 || keyLength == 24 || keyLength == 32
        } catch (e: IllegalArgumentException) {
            println("Failed to decode key❗: " + e.message)
            return false
        }
    }

    @JvmStatic
    fun decodeKey(encodedKey: String?): SecretKey {
        require(isValidAESKey(encodedKey)) { "Invalid AES key❗." }
        val decodedKey = Base64.getDecoder().decode(encodedKey)
        return SecretKeySpec(decodedKey, "AES")
    }

    @JvmStatic
    fun validateKey(key: String?) {
        require(isValidAESKey(key)) { "Invalid AES key. Ensure it is 128, 192, or 256 bits." }
    }
}
